tests/integration/new_relations/helpers.py

```diff
@@ -39,12 +48,25 @@ async def build_connection_string(
     # Get the connection data exposed to the application through the relation.
     if database is None:
         database = f'{application_name.replace("-", "_")}_{relation_name.replace("-", "_")}'
-    username = await get_application_relation_data(
-        ops_test, application_name, relation_name, "username", relation_id, relation_alias
-    )
-    password = await get_application_relation_data(
-        ops_test, application_name, relation_name, "password", relation_id, relation_alias
-    )
+
+    if secret_uri := await get_application_relation_data(
```

Comment:
Always check for secret uri, so that we don't have to keep track of charms that have not switched to the secrets implementation (data-integrator).